### PR TITLE
Prohibit file overwrite in iRODS

### PR DESCRIPTION
**Pull request by alvacouch**
@hyi @pkdash This simple PR allows Django to overwrite existing files, thus preserving its ability to name files the way people request.

### CODE REVIEW

**Comment by hyi** [comment]
@alvacouch +1. You can merge it, then update SHA reference to the new django_irods submodule in the corresponding hydroshare branch that allows django to overwrite files, in order for the updated django_irods submodule to be referenced by hydroshare repo. You can ask @mjstealey to help out with this if needed.

**Comment by alvacouch** [comment]
@hyi I just realized that we decided to do something different. In HydroShare issue 1744, we decided to raise an exception if a duplicate is proposed, and force us to do an explicit delete. 

Will change this PR accordingly and notify you.

**Comment by alvacouch** [comment]
@pkdash This PR will definitely break the old file handling. I needed to make rather extensive changes in HydroShare branch 1710 to get it to work properly. Merging should be delayed until that PR is complete and ready to merge.

**Comment by pkdash** [comment]
@alvacouch Sounds good.

**Comment by hyi** [comment]
@alvacouch There are conflicts that need to be resolved for this PR.

**Comment by alvacouch** [comment]
@pkdash This is ready to merge and merge is necessary in order to finish the HydroShare PR.

File: views.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
         pre_check_bag_flag.send(sender=resource_cls, resource=res)
         if bag_modified == "true":
             if metadata_dirty == 'true':
-                create_bag_files(res, fed_zone_home_path=res.resource_federation_path)
+                create_bag_files(res)
```
Comment (pkdash) [inline views.py]: I assume this change will be reflected in the hydroshare PR.

**Comment by pkdash** [comment]
@alvacouch +1 (left one comment though).

File: views.py (hunk 1 above)
Comment (alvacouch) [inline views.py]: Yes. That argument Is redundant. It has been eliminated.

**Pull request merged**